pkg/cloudprovider/cloudprovider.go

```diff
@@ -58,20 +58,20 @@ func GetCloudConfig(pconfig providerconfigtypes.Config, kubeletVersion string) (
 	return "", errors.New("unknown cloud provider")
 }
 
-func KubeletCloudProviderName(cloudProvider providerconfigtypes.CloudProvider) (osmv1alpha1.CloudProvider, error) {
+func KubeletCloudProviderConfig(cloudProvider providerconfigtypes.CloudProvider) (inTreeCCM bool, external bool, err error) {
 	switch osmv1alpha1.CloudProvider(cloudProvider) {
 	case osmv1alpha1.CloudProviderAWS, osmv1alpha1.CloudProviderAzure, osmv1alpha1.CloudProviderGoogle,
 		osmv1alpha1.CloudProviderOpenstack, osmv1alpha1.CloudProviderVsphere:
-		return osmv1alpha1.CloudProvider(cloudProvider), nil
+		return true, false, nil
```

Comment:
Openstack and vSphere make use of the out of tree CCM, I think that they should be moved in the following case statement, along with Kubevirt.

Reply:
Here we are checking if the cloud provider has an in-tree solution or not. If someone wants to use out of tree CCM then they have to set the `external-provider` flag for OSM.

Reply:
I have just refactored the return values etc but the overall functionality remains the same. This is the relevant change from machine-controller that required this, for context.

https://github.com/kubermatic/machine-controller/pull/1162/files

But I didn't want to introduce a separate field for `KubeletCloudProviderName` so just went with `InTreeCCMAvailable`; the actual thing that we are checking here.